### PR TITLE
Add a CSV relation writer.

### PR DESCRIPTION
**Pull request by blambeau**

### CODE REVIEW

**Comment by blambeau** [comment]
@ygu @felixyz @OSdave 

I'd like everything related to exporting data @ Enspirit & Klaro to use relational algebra (hence Bmg) as much as possible.
I start a first writer here, comments are welcome.

Every contribution to export relations as HTML, XLS(X) are more than welcome next. Please synchronize with me.

File: lib/bmg/relation.rb [hunk 1]
```diff
@@ -105,6 +105,19 @@ def to_json(*args, &bl)
...
+    # Writes the relation data to CSV.
+    #
+    # `output` and `options` are what CSV::new itself
+    # recognizes, default options are CSV's.
+    #
+    # When no output is used, the method uses a string.
+    #
+    # The method always returns the output.
+    def to_csv(options = {}, output = nil)
+      options, output = {}, options unless options.is_a?(Hash)
```
Comment (felixyz) [inline lib/bmg/relation.rb:117]: Maybe use the same arg names as `CSV.new`? So `string_or_io` instead of `output`.

File: lib/bmg/writer/csv.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      DEFAULT_OPTIONS = {
+      }
+
+      def initialize(options)
+        @options = DEFAULT_OPTIONS.merge(options)
+      end
+      attr_reader :options
+
+      def call(r, output = nil)
```
Comment (felixyz) [inline lib/bmg/writer/csv.rb:14]: Is `r` used consistently as a variable name for relations? If not, IMO it's unnecessarily terse, I would prefer `relation` or at least `rel`.

File: spec/unit/writer/test_csv.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          {id: "1", name: "Bernard Lambeau"},
+          {id: "2", name: "Yoann;Guyot"}
+        ]
+      }
+
+      context 'with default options and no output' do
+        let(:args){ [] }
+
+        it 'works' do
+          expected = <<-CSV.gsub(/^\s+/, "")
```
Comment (felixyz) [inline spec/unit/writer/test_csv.rb:19]: Unless I'm missing something, you should be able to use the squiggly heredoc `<<~` here and drop the `gsub`.

File: spec/unit/writer/test_csv.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          expect(subject).to eql(args[1])
+          expected = <<-CSV.gsub(/^\s+/, "")
+            1,Bernard Lambeau
+            2,Yoann;Guyot
+          CSV
+          expect(subject.string).to eql(expected)
+        end
+      end
+
+      context 'specifying inline CSV options' do
```
Comment (felixyz) [inline spec/unit/writer/test_csv.rb:53]: This example would be more valuable if it specified any other `col_sep` than `;`.

**Comment by felixyz** [comment]
Nice! It would be natural for the next export to separate the extraction of headers and attributes from the building of the new output. I'd be happy to implement one or two more output formats, like XLS. Is there are card for this in Klaro?

**Comment by ygu** [comment]
That's great indeed ! Everything ok to me.
Will make sure to add additionnal writers too.

**Comment by blambeau** [comment]
Thank you @ygu @felixyz.

@felixyz I rarely have klaro boards for open-source projects (maybe we should).

For me, "adding a .xls writer to Bmg" is as a natural step for a user story saying "I'd like to export a Klaro board to Excel". Buy I should certainly make this kind of strategy (continuously nuturing open-source libraries as implementation strategies for non open-source software) much clearer for the team.

**Pull request merged**